cg/cli/workflow/nf_analysis.py

```diff
@@ -124,3 +130,33 @@ def report_deliver(context: CGConfig, case_id: str, dry_run: bool) -> None:
     except Exception as error:
         LOG.error(f"Could not create report file: {error}")
         raise click.Abort()
+
+
+@click.command("store-housekeeper")
+@ARGUMENT_CASE_ID
+@OPTION_DRY
+@click.pass_obj
+def store_housekeeper(context: CGConfig, case_id: str, dry_run: bool) -> None:
+    """Store a finished RNAFUSION and TAXPROFILER analysis in Housekeeper and StatusDB."""
+    analysis_api: NfAnalysisAPI = context.meta_apis[MetaApis.ANALYSIS_API]
+    housekeeper_api: HousekeeperAPI = context.housekeeper_api
+    status_db: Store = context.status_db
+
+    try:
+        analysis_api.status_db.verify_case_exists(case_internal_id=case_id)
+        analysis_api.trailblazer_api.is_latest_analysis_completed(case_id=case_id)
+        analysis_api.verify_deliverables_file_exists(case_id=case_id)
+        analysis_api.upload_bundle_housekeeper(case_id=case_id, dry_run=dry_run)
+        analysis_api.upload_bundle_statusdb(case_id=case_id, dry_run=dry_run)
+        analysis_api.set_statusdb_action(case_id=case_id, action=None, dry_run=dry_run)
+    except ValidationError as error:
+        LOG.warning("Deliverables file is malformed")
+        raise error
+    except CgError as error:
+        LOG.error(f"Could not store bundle in Housekeeper and StatusDB: {error}")
+        raise click.Abort()
+    except Exception as error:
+        LOG.error(f"Could not store bundle in Housekeeper and StatusDB: {error}!")
+        housekeeper_api.rollback()
+        status_db.session.rollback()
+        raise click.Abort()
```

Comment:
Can you extract this logic from the CLI layer into the API and make a call to a single function that performs the relevant functionality.

Ideally the function will except possible errors and propagates a single error to the CLI layer that gets excepted.

Reply:
Example of what I described above:

```Python
@balsamic.command("store")
@ARGUMENT_CASE_ID
@DRY_RUN
@click.pass_context
def store(context: click.Context, case_id: str, dry_run: bool):
    """Generate Housekeeper report for CASE ID and store in Housekeeper"""
    LOG.info(f"Storing analysis for {case_id}")
    report_and_store_api = ReportAndStoreAPI(context)
    try:
        report_and_store_api.store_and_deliver(case_id=case_id, dry_run=dry_run)
    except (DeliverError, StoreError) as error:
        LOG.error(f"An error occurred: {error}.")
```

Reply:
I was thinking of moving this into `cg/meta/workflow/nf_analysis.py`.

Reply:
@ChrOertlin Updated the PR.